src/loader/tr5level.cpp

```diff
@@ -0,0 +1,293 @@
+    if(comp_size > 0)
+    {
+        if(m_textile32.empty())
+        {
+            std::vector<uint8_t> uncomp_buffer(uncomp_size);
+
+            m_textile16.resize(numTextiles);
+            std::vector<uint8_t> comp_buffer(comp_size);
+
+            m_reader.readBytes(comp_buffer.data(), comp_size);
+
+            uLongf size = uncomp_size;
+            if(uncompress(uncomp_buffer.data(), &size, comp_buffer.data(), comp_size) != Z_OK)
+                throw std::runtime_error("read_tr5_level: uncompress");
+
+            if(size != uncomp_size)
+                throw std::runtime_error("read_tr5_level: uncompress size mismatch");
+
+            SDL_RWops* newsrcSDL = SDL_RWFromMem(uncomp_buffer.data(), uncomp_size);
+            if(newsrcSDL == nullptr)
+                throw std::runtime_error("read_tr5_level: SDL_RWFromMem");
+
+            io::SDLReader newsrc(newsrcSDL);
+            newsrc.readVector(m_textile16, numTextiles - numMiscTextiles, &WordTexture::read);
+        }
+        else
+        {
+            m_reader.skip(comp_size);
+        }
+    }
+
+    uncomp_size = m_reader.readU32();
+    if(uncomp_size == 0)
+        throw std::runtime_error("read_tr5_level: textiles32d uncomp_size == 0");
+
+    comp_size = m_reader.readU32();
+    if(comp_size > 0)
+    {
+        if(m_textile32.empty())
+        {
+            std::vector<uint8_t> uncomp_buffer(uncomp_size);
+
+            if((uncomp_size / (256 * 256 * 4)) > 3)
+                std::cerr << "read_tr5_level: num_misc_textiles > 3\n";
+
+            std::vector<uint8_t> comp_buffer(comp_size);
+            m_reader.readBytes(comp_buffer.data(), comp_size);
+
+            uLongf size = uncomp_size;
+            if(uncompress(uncomp_buffer.data(), &size, comp_buffer.data(), comp_size) != Z_OK)
+                throw std::runtime_error("read_tr5_level: uncompress");
+
+            if(size != uncomp_size)
+                throw std::runtime_error("read_tr5_level: uncompress size mismatch");
+
+            SDL_RWops* newsrcSDL = SDL_RWFromMem(uncomp_buffer.data(), uncomp_size);
+            if(newsrcSDL == nullptr)
+                throw std::runtime_error("read_tr5_level: SDL_RWFromMem");
+
+            io::SDLReader newsrc(newsrcSDL);
+            newsrc.readVector(m_textile32, numTextiles - numMiscTextiles, &DWordTexture::read);
+        }
+        else
+        {
+            m_reader.skip(uncomp_size);
+        }
+    }
+
+    // flags?
+    /*
+       I found 2 flags in the TR5 file format. Directly after the sprite textures are 2 ints as a flag. The first one is the lara type:
+       0 Normal
+       3 Catsuit
+       4 Divesuit
+       6 Invisible
+
+       The second one is the weather type (effects all outside rooms):
+       0 No weather
+       1 Rain
+       2 Snow (in title.trc these are red triangles falling from the sky).
+     */
+#if 1
+    m_reader.skip(2*sizeof(uint16_t));
+#else
+    i = m_src.readU16();
+    i = m_src.readU16();
+#endif
```

Comment:
Can you please integrate this into TR_Level structure? Seems these two values are ignored. You can create global flags for level structure, one called `m_LaraType` and another called `m_WeatherType`.

Actually, `m_WeatherType` can be also re-used for TR3, where weather was hardcoded.